libutils/buffer.c

```diff
@@ -445,6 +445,33 @@ void BufferSetMode(Buffer *buffer, BufferBehavior mode)
     buffer->mode = mode;
 }
 
+Buffer* BufferFilter(Buffer *buffer, FilterFn filter, const bool invert)
+{
+    assert(buffer);
+
+    Buffer *filtered = BufferNew();
+    for (unsigned int i = 0; i < buffer->used; ++i)
+    {
+        const bool test = (*filter)(buffer->buffer[i]);
+
+        if (test || (invert && !test))
```

Comment:
This will not invert, it will simply include every character if ``invert`` is true, in which case the expression essentially becomes ``test || !test``. A better expression would be ``invert ? !test : test``.